src/main.cc

```diff
@@ -2,26 +2,30 @@
 #include <memory>
 
 #include "gmsh.h"
+#include "io.h"
 
 int main(int argc, char** argv) {
   try {
     if (argc != 3) {
-      std::cout << "Usage: ./mpm-point-generator /path/to/mesh_file.msh\t"
+      std::cout << "Usage: ./mpm-point-generator /path/to/path_file.txt\n"
                    "Dimension 2 or 3\n";
       throw std::runtime_error("Incorrect number of input arguments");
     }
+    //! Get ndimension
+    const unsigned ndimension = std::atoi(argv[2]);
 
-    int dimension = std::atoi(argv[2]);
-    std::string filename = argv[1];
+    //! Get path file location including directory
+    const std::string pathfile_name = argv[1];
```

Comment:
You don't need the variable `pathfile_name`, remove it altogether and use `argv` directly.